environment/commands/transfer-eth-erc20-from-near.js

```diff
@@ -0,0 +1,216 @@
+        web3.eth.accounts.wallet.add(ethMasterAccount);
+        web3.eth.defaultAccount = ethMasterAccount.address;
+        ethMasterAccount = ethMasterAccount.address;
+        let clientContract = new web3.eth.Contract(
+            // @ts-ignore
+            JSON.parse(fs.readFileSync(RainbowConfig.getParam('near2eth-client-abi-path'))),
+            RainbowConfig.getParam('near2eth-client-address'), {
+                from: ethMasterAccount,
+                handleRevert: true,
+            },
+        );
+
+        let clientBlock;
+        let clientBlockHeight;
+        let clientBlockValidAfter;
+        let clientBlockHashB58;
+        let clientBlockHashHex;
+        let clientBlockMerkleRoot;
+        while (true) {
+            clientBlock = await clientContract.methods.last().call();
+            clientBlockHeight = new BN(clientBlock.height);
+            clientBlockValidAfter = new BN(clientBlock.validAfter);
+            clientBlockHashHex = await clientContract.methods.blockHashes(clientBlockHeight).call();
+            clientBlockMerkleRoot = await clientContract.methods.blockMerkleRoots(clientBlockHeight).call();
+            clientBlockHashB58 = bs58.encode(toBuffer(clientBlockHashHex));
+            console.log(`Current light client head is: hash=${clientBlockHashB58}, height=${clientBlockHeight.toString()}`);
+
+            const chainBlock = await web3.eth.getBlock('latest');
+            const chainBlockTimestamp = new BN(chainBlock.timestamp);
+            if (clientBlockHeight.gt(outcomeBlockHeight)) {
+                console.log(`Near2EthClient block is at ${clientBlockHeight.toString()} which is further than the needed block ${outcomeBlockHeight.toString()}`);
+                break;
+            } else if (chainBlockTimestamp.gte(clientBlockValidAfter) && clientBlockHeight.eq(outcomeBlockHeight)) {
+                console.log(`Near2EthClient block is at ${clientBlockHeight.toString()} which is the block of the outcome. And the light client block is valid.`);
+                break;
+            } else if (chainBlockTimestamp.lt(clientBlockValidAfter) && clientBlockHeight.eq(outcomeBlockHeight)) {
+                const sleepSec = clientBlockValidAfter.sub(chainBlockTimestamp).toNumber();
+                console.log(`Block ${clientBlockHeight.toString()} is not valid yet. Sleeping ${sleepSec} seconds.`);
+                await sleep(sleepSec * 1000);
+            } else {
+                const sleepSec = 10;
+                console.log(`Block ${outcomeBlockHeight.toString()} is not available on the light client yet. Current height of light client is ${clientBlockHeight.toString()}. Sleeping ${sleepSec} seconds.`);
+                await sleep(sleepSec * 1000);
+            }
+        }
+
+        console.log(`Burnt ${JSON.stringify(command.amount)}`);
+        const new_balance = await nearTokenContract.get_balance({
```

Comment:
I feel we'll need some more error recovering. For example, what if burn is success but get errors in following steps? Should we
- revert the burnt after sometime,
- or provide user instructions to continue transfer with a valid burn?

Reply:
Yes, we should provide user instructions on how to recover the operation. I haven't added it yet. You could add it too :)